### ISSUE TITLE
Re-vamp the "invite members" functionality

### ISSUE THREAD

**Issue by mmbotelho** [okTurtles/group-income#553]
_**Note**: I decided to create this issue to replace the issue #406 - I did this because the functionality changed significantly, and the problems originally discussed there no longer make sense._

## Problem
**Right now, there's 2 ways to add members to a group:**

- When creating a group;
- On the group dashboard.

![Captura de ecrã 2019-04-13, às 19 11](https://user-images.githubusercontent.com/22004566/56083720-c777ea80-5e20-11e9-88c5-e46949a33a28.png)
![Captura de ecrã 2019-04-13, às 19 11-1](https://user-images.githubusercontent.com/22004566/56083719-c777ea80-5e20-11e9-910e-8cab09a55d57.png)

**There are several problems with the way we are currently doing this:**
- The user adding a new member needs to know their EXACT username - this can create a lot of friction;
- Having this step on the group creation makes it unnecessarily long and time-consuming;
- It's still not possible to create or vote on a proposal to add a new member;

## Solution
Now, the user can only add members on the dashboard page. It is **no longer possible** to do so in the group creation flow (so this step should be removed from the group creation flow). 

The mechanism to add new members is fundamentally different now - we no longer expect the user to know the username of the new member; instead, we provide them with a unique access link, that the other person can use as a "key" to join the group.

There are now 2 ways of joining a group:

- **Without a voting process** - this is true when the group has been created for **less than 1 week** AND/OR has **less than 3 members**. In this case there's one link that can be used by up to 60 people (during that first week; after that, the link should expire).
- **With a voting process** - when the group has more than 1 week and/or more than 3 members.

When a voting process is necessary, a proposal should be created (one for each member added) and only after it has been accepted, the access link is generated. Since we have the pull request #343 (Proposals / voting logic) still open, I will not consider that part of the work necessary to complete this card. So, in short, everything that happens after a proposal is created is not part of this card.

Check out the Figma mockups here  [https://www.figma.com/file/ZR0DRPhhUuHdJtf7PbGAM6tW/Group-Income?node-id=875%3A2](https://www.figma.com/file/mxGadAHfkWH6qApebQvcdN/Group-Income-2.0?node-id=875%3A2)

Let me know if you have any questions!

**Comment by taoeffect** [comment]
> When a voting process is necessary, a proposal should be created (one for each member added) and only after it has been accepted, the access link is generated.

Yep! And I should add that generated links (when people are being invited via a proposal) are tied to the name of the person being invited, and the UI should somehow reflect this so that Bob's link is paired with Bob, and Susan's link is paired with Susan, so that Bob's link doesn't get sent to Susan. The UI should make it very clear, in other words, who each invite link belongs to, so the wrong link isn't sent to the wrong person.

**Comment by mmbotelho** [comment]
The issues you mentioned above were all solved on the Dashboard mockups, right?

Once a proposal is accepted, the link is displayed on the UI - [here's a refresher](https://www.figma.com/file/ZR0DRPhhUuHdJtf7PbGAM6tW/Group-Income?node-id=1579%3A202)

**Comment by mmbotelho** [comment]
PR #655 should solve this issue

**Comment by sandrina-p** [comment]
@mmbotelho no it doesn't solve, but it's related only in the part of creating the layout for the invitation by link widget + modal.
The full functionality will be addressed in another PR.

---

Also, could you update the Figma file?

**Comment by sandrina-p** [comment]
Also, could you update the Figma file too?

**Comment by mmbotelho** [comment]
Updated @sandrina-p 😄

**Comment by mmbotelho** [comment]
Just a heads up @sandrina-p 
<img width="1010" alt="Captura de ecrã 2019-10-16, às 08 50 24" src="https://user-images.githubusercontent.com/22004566/66891499-02d2bc00-eff2-11e9-8a34-cb1b203a78e0.png">

I reduced the copy of the "Share this link..." so it would fit in one line on desktop (as discussed on the hackathon)

**Comment by mmbotelho** [comment]
I believe your latest PR closes this issue, right @sandrina-p?

I am closing it because I'm 99% sure it does. Feel free to open it if not.
